### PR TITLE
2022 Benchmarking and Algorithm Fixes (Linted)

### PR DESCRIPTION
**Pull request by ayushchakra**

### CODE REVIEW

File: benchmarking/datasets/desnat_straight_once.json
Comment (ayushchakra) [inline benchmarking/datasets/desnat_straight_once.json:1]: @richardli03 @ruchadave16 Do these jsons needs to be checked into the repo?

File: configs/ground_truth_mapping.json [hunk 1]
```diff
@@ -0,0 +1,326 @@
+{
+  "mac_1_2": [
```
Comment (ayushchakra) [inline configs/ground_truth_mapping.json:2]: maybe this shared config file was a bad idea.....

File: ground_truth/DESNAT_RPY.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+rpy 23 1.169040 -0.229212 0.412466 -1.603438 0.003699 1.558692
+rpy 22 -2.710969 -4.367945 0.369046 -1.576496 -0.006221 -1.588495
```
Comment (ayushchakra) [inline ground_truth/DESNAT_RPY.txt:2]: Are these rtabmap files still relevant? The tags aren't up anymore

File: configs/firebase_device_config.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (richardli03) [inline configs/firebase_device_config.json:1]: actually super super useful thank you!

File: configs/ground_truth_mapping.json (hunk 1 above)
Comment (richardli03) [reply configs/ground_truth_mapping.json:2]: I kind of see why you might think that... but we're never looking at this right so it's maybe okay?

File: ground_truth/DESNAT_RPY.txt (hunk 2 above)
Comment (richardli03) [reply ground_truth/DESNAT_RPY.txt:2]: nope, none of these .txts are relevant. Maybe we should delete?

File: map_processing/graph_vertex_edge_classes.py
Comment (richardli03) [inline map_processing/graph_vertex_edge_classes.py:1]: Will be honest, I'm looking over these but most of it is a bit over my head. Will assume it's mainly lint fixes though.

**Review by richardli03 — CHANGES_REQUESTED** [review]
Changes to make: 
- Delete .txt files from RTABmap
- wait for pruvolo's opinions on benchmarking folder, maybe iros folder?

**Comment by ayushchakra** [comment]
@richardli03 let me merge this pls so its in master for summer research

File: .gitignore [hunk 4]
```diff
@@ -22,4 +23,9 @@ venv
 
 /saved_sweeps/**/*sweep.*
 
-.vscode/
```
[the remark this answers is not in the document]
Comment (ayushchakra) [reply .gitignore:25]: yea

[the remark this answers is not in the document]
Comment (ayushchakra) [reply .gitignore:25]: oh we 100% should be ignoring them

File: archive/iros_2023_data/experiment_one_mac_fixed.ipynb
[the remark this answers is not in the document]
Comment (ayushchakra) [reply archive/iros_2023_data/experiment_one_mac_fixed.ipynb:1]: I feel like they do - represents the current state of the project

File: archive/iros_2023_data/run.py [hunk 5]
```diff
@@ -0,0 +1,334 @@
+import os
```
[the remark this answers is not in the document]
Comment (ayushchakra) [reply archive/iros_2023_data/run.py:1]: tbh I don't think anyone looks at the archive folder

File: benchmarking/datasets/floor_2_obleft.json
[the remark this answers is not in the document]
Comment (ayushchakra) [reply benchmarking/datasets/floor_2_obleft.json:1]: We could create another bucket in firebase

File: configs/ground_truth_mapping.json [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+    "generated_23-02-23-01-07-34",
+    "generated_23-02-23-01-11-12",
+    "generated_23-02-23-01-19-23",
+    "generated_23-02-23-01-19-32",
+    "generated_23-02-23-01-19-41",
+    "generated_23-02-23-01-22-05",
+    "generated_23-02-23-01-23-36",
+    "generated_23-02-23-01-23-46",
+    "generated_23-02-23-01-24-00",
+    "generated_23-02-23-01-25-40",
```
[the remark this answers is not in the document]
Comment (ayushchakra) [reply configs/ground_truth_mapping.json:240]: yeaaaaaaaa checking this in was also a bad idea

File: archive/repeat_run_automation.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import os
```
[the remark this answers is not in the document]
Comment (ayushchakra) [reply archive/repeat_run_automation.py:1]: `run.py` abstracts out the stuff in optimize graphs and manage cache into functions that can be run in other contexts, this file automizes the process of generating files and running parameter sweeps on them

**Review by redhotchili0909 — APPROVED** [review]

File: .gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (paulruvolo) [reply .gitignore:25]: Yes, please make that change.

File: archive/iros_2023_data/experiment_one_mac_fixed.ipynb
[the remark this answers is not in the document]
Comment (paulruvolo) [reply archive/iros_2023_data/experiment_one_mac_fixed.ipynb:1]: I would like to push the IROS ipynbs.  I think that is good to have front and center (i.e., not in a branch).

**Pull request merged**